raft.c

```diff
@@ -2346,6 +2361,89 @@ void applyShardGroupChange(RedisRaftCtx *rr, raft_entry_t *entry)
     }
 }
 
+void replaceShardGroups(RedisRaftCtx *rr, raft_entry_t *entry)
+{
+    // 1. reset sharding info
+    ShardingInfo *si = rr->sharding_info;
+
+    ShardGroup *local = NULL;
+    if (si->shard_group_map != NULL) {
+        RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(si->shard_group_map, "^", NULL, 0);
+
+        char *key;
+        size_t key_len;
+        ShardGroup *data;
+
+        while ((key = RedisModule_DictNextC(iter, &key_len, (void **) &data)) != NULL) {
+            /* local shardgroup will not have a filled in id */
+            if (*data->id != 0) {
+                RedisModule_Free(data);
```

Comment:
Similar here probably, need to call ShardGroupFree() ?

Reply:
yes and another place too I found